readme.md

```diff
@@ -38,10 +41,29 @@ The source object to copy properties from.
 
 #### mapper
 
-Type: `(sourceKey, sourceValue, source) => [targetKey, targetValue, mapperOptions?] | mapObjectSkip`
+Type: `(sourceKey, sourceValue, source, path) => [targetKey, targetValue, mapperOptions?] | mapObjectSkip`
 
 A mapping function.
 
+##### path
+
+Type: `string[]`
+
+When using `deep: true`, this is the sequence of keys to reach the current value from the `source`, otherwise it is an empty array.
+
+For arrays, the key is the index of the element being mapped.
+
+```js
+import mapObject from "map-obj";
+
+const object = {foo: {bar: [2], baz: [1, 2, 3]}}
+const mapper = (key, value, source, path) => path.join(".") === "foo.baz" ? [key, 3] : [key, value];
+const result = mapObject(object, 	mapper, { deep: true });
+
+console.log(result);
+//=> {foo: {bar:[2], baz: [3, 3, 3]}}
+```
```

Comment:
Readme and index.d.ts should be in sync.

Reply:
I took this to mean you want the `path` annotated in the definition file, let me know if I misunderstood.